### ISSUE TITLE
[Bug] getAssetListPaged empty 

### ISSUE THREAD

**Issue by scuti-dai** [fluttercandies/flutter_photo_manager#921]
### Version

2.6.0

### Platforms

iOS

### Device Model

Iphone 5 (ios 13.6.1)

### flutter info

```sh
Flutter (Channel stable, 3.7.7, on macOS 13.0 22A380 darwin-arm64, locale en)
    • Flutter version 3.7.7 on channel stable at /opt/homebrew/Caskroom/flutter/2.10.3/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ad6cd72c0 (5 weeks ago), 2023-03-08 09:41:59 -0800
    • Engine revision 1837b5be5f
    • Dart version 2.19.4
    • DevTools version 2.20.1
```

### How to reproduce?

 final result = await albums.first.getAssetListPaged(page: 0, size: 100);

- albums is list data but when i call function getAssetListPaged. it return empty
But OS > 15 it work fine
- if i used albums[1] it work well too.

![Screen Shot 2023-04-11 at 17 33 14](https://user-images.githubusercontent.com/104118833/231138463-9d842ffc-23ee-4127-a533-25854b5ef624.png)

**Comment by delfme** [comment]
I get empty results only for following albums:
- Selfies
- Live Photos
- Protraits
- Long Exposure
- Panoramas
- Time-lapse
- Slo-mo

Looking at the names, it seems like the issues affect only some special albums. Please notice Im using a fake thumb for the corrupted results.

![Simulator Screen Shot - iPhone 13 Pro - 2023-04-16 at 17 40 31](https://user-images.githubusercontent.com/53510751/232324258-896b10b1-a19d-4026-b0b0-01c3276a8a7c.png)

I also don't get correct assetCount for "all" albums.

**Comment by delfme** [comment]
Should have I to file a different PR to report issue with `assetCount`?

**Comment by scuti-dai** [comment]
i fixed this bug like:

    final List<AssetEntity> results =
        await albums.first.getAssetListPaged(page: page, size: 100);
    final firstName = albums.first.name;
    const recentName = AppParamConstants.albumRecent;
    const rollName = AppParamConstants.albumCameraRoll;
    final notRecentAlbum = firstName != recentName && firstName != rollName;
    if (GetPlatform.isIOS && (results.isEmpty || notRecentAlbum)) {
      for (int index = 0; index < albums.length; index++) {
        final item =
            await albums[index].getAssetListPaged(page: page, size: 100);
        if (albums[index].name == recentName) {
          results.addAll(item);
          break;
        }
        if (albums[index].name == rollName) {
          results.addAll(item);
          break;
        }
      }
    }

**Comment by tomoyukitanaka1171** [comment]
@delfme 
Maybe, you should set following option. 

```
imageOption: const FilterOption(
      sizeConstraint: SizeConstraint(ignoreSize: true), // If you get large size Image, you should ignore.
      onlyLivePhotos: false, // default false, but maybe you set true.
    ),
```

Or, maybe your RequestType is wrong.
If you wanna obtain image, you should set it.

```
  final RequestType requestType = RequestType.image;
```

I would appreciate it if you could help me.

**Comment by delfme** [comment]
HI @tomoyukitanaka1171 thx for the kind attention.

My function is meant to just fetch the thumbnail of each album and this is the code
```
Map<AssetPathEntity, Uint8List?> albums = {};

// Get thumb for album
Future<Uint8List?> _getFirstThumb(AssetPathEntity album) async {
  var assetList;
  assetList = await album.getAssetListRange(
    start: 0,
    end: 1,
  );
  return assetList.first
      .thumbnailDataWithSize(const ThumbnailSize(500, 500), quality: 25);
}


/// Load all the albums in the device
Future<Map<AssetPathEntity, Uint8List?>> loadAlbums() async {
  // ignore: no_leading_underscores_for_local_identifiers
  final _albums = await PhotoManager.getAssetPathList(type: RequestType.common);

  _albums.sort((a, b) {
    if (a.isAll) return -1;
    if (b.isAll) return 1;
    return 0;
  });

  final result = <AssetPathEntity, Uint8List?>{};

  for (final album in _albums) {
    final thumb = await _getFirstThumb(album);
    result.addAll({album: thumb});
  }

  albums = result;

  if (medias.isEmpty) {
    await getPhotos(
      _albums.first,
      pageIndex: 0,
    ).toList().then((photos) {
      if (medias.isEmpty) {
        medias.addAll(photos);
        return;
      }
    });
  }

  return result;
}
```

It has always worked fine.  Can you suggest how to change it for a try?

**Comment by delfme** [comment]
Log issue point here coz for some album it return nothing.

```
  assetList = await album.getAssetListRange(
    start: 0,
    end: 1,
  );
  return assetList.first
      .thumbnailDataWithSize(const ThumbnailSize(500, 500), quality: 25);
```

**Comment by tomoyukitanaka1171** [comment]
@delfme
You're welcome.

Thanks for the information.

I don't know, but Maybe Images size that you wanna get are over 100000pixel ( width or height ).

Default, SizeConstraint set maxWidth and maxHeight it.

```
  const SizeConstraint({
    this.minWidth = 0,
    this.maxWidth = 100000,
    this.minHeight = 0,
    this.maxHeight = 100000,
    this.ignoreSize = false,
  });
```

So, in that case, you should set `ignoreSize = true`.

Can you try following ?

```
final FilterOptionGroup _filterOption = FilterOptionGroup(
    imageOption: const FilterOption(
      sizeConstraint: SizeConstraint(ignoreSize: true),
    ),
  );

final _albums = await PhotoManager.getAssetPathList(type: RequestType.common, filterOption: _filterOption);
```

 
I hope you will find it useful.

**Comment by delfme** [comment]
@tomoyukitanaka1171 just tried it. This is error stack:
```
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: Bad state: No element
#0      List.first (dart:core-patch/growable_array.dart:343:5)
#1      _getFirstThumb (package:presence/screens/story_media_selector/album_selector.dart:50:20)
<asynchronous suspension>
#2      loadAlbums (package:presence/screens/story_media_selector/album_selector.dart:87:19)
<asynchronous suspension>
```

The error points to this line from above func `_getFirstThumb(AssetPathEntity album)`
```
 return assetList.first
      .thumbnailDataWithSize(const ThumbnailSize(500, 500), quality: 25);
```

**Comment by delfme** [comment]
If you can add logs to a 'debug-branch' I will try again with it and revert.
I'm using ios15.5 and ios16.1, both emulator and physical device

**Comment by delfme** [comment]
@CaiJingLong @tomoyukitanaka1171 just noticed this fix is planned for 2.6.1.  Is there a workaround I could use mentime or do I just have to downgrade?

**Comment by iosephmagno** [comment]
I can workaround this issue by using photo_manager: 2.1.2 , but only on physical device (emulator stalls with no errors).

It looks like issue is caused by await `album.assetCountAsync` that never return a value.

On 2.1.2 we have old album.assetCount which works fine.
I have 41.490 images in Recents album and I think this might play a role in the issue.
Idk why 2.1.2 doesn't work on emulator. But it looks like some iOS special album crashes `PhotoManager.getAssetPathList()`

Tested on iphone ios 16.1 and emulator 16 and 15.5

**Comment by iosephmagno** [comment]
It also looks like on emulator the methods to get thumbnail (thumbnailDataWithSize, thumbnailDataWithOption and thumbnailData) fail to return a value and throw no errors.

**Comment by AlexV525** [comment]
From the above comments, there're multiple issues that are not all related to `getAssetListPaged`.

>albums is list data but when i call function getAssetListPaged. it return empty
But OS > 15 it work fine if i used albums[1] it work well too.

@scuti-dai The first album is `Animated`, which might not contain any assets. `albums[1]` is another album that is different from `albums.first`, it's reasonable that your first album has empty assets but the second one has some.

> I also don't get correct assetCount for "all" albums.

@delfme The `assetCount` has already been deprecated so you'll need to move on to the `assetCountAsync`.

> just noticed this fix is planned for 2.6.1. Is there a workaround I could use mentime or do I just have to downgrade?

The relevant exception should not occur in 2.6.0, which means you should avoid using the main branch of our code unless you're intended to.

> On 2.1.2 we have old album.assetCount which works fine.
I have 41.490 images in Recents album and I think this might play a role in the issue.
Idk why 2.1.2 doesn't work on emulator. But it looks like some iOS special album crashes `PhotoManager.getAssetPathList()`

@iosephmagno Please provide logs and your question is out of the scope of the current issue, consider submitting a new one with the full info provided.

**Comment by iosephmagno** [comment]
Hi @AlexV525 got to fix my issue

**Comment by CaiJingLong** [comment]
> @CaiJingLong @tomoyukitanaka1171 just noticed this fix is planned for 2.6.1...

@delfme

If it can be reproduced, it can be planned to fix it in 2.6.1, but if no more information can be provided, the plan needs to be postponed.

If possible, you need to provide the following information:
1. The complete code to reproduce, preferably a Widget, with no references to other files except photo_manager.
2. The complete environment to reproduce, a brand new iOS simulator, and then I need how to add albums or pictures to reproduce your environment.

**Comment by CaiJingLong** [comment]
No more information to help reproduce, so remove it from milestone.
